File: app/src/main/java/com/example/nasa/data/model/APODRoom.java
package com.example.nasa.data.model;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.jetbrains.annotations.NotNull;

@Entity
public final class APODRoom {

    @PrimaryKey
    @ColumnInfo(name = "date")
    @NonNull
    private final String mDate;

    @ColumnInfo(name = "explanation")
    @NonNull
    private final String mExplanation;

    @ColumnInfo(name = "title")
    @NonNull
    private final String mTitle;

    @ColumnInfo(name = "url")
    @NonNull
    private final String mUrl;

    @ColumnInfo(name = "copyright")
    @NonNull
    private final String mCopyright;


    public APODRoom(@NonNull String date, @NonNull String explanation, @NonNull String title, @NonNull String url, @Nullable String copyright) {
        mDate = date;
        mExplanation = explanation;
        mTitle = title;
        mUrl = url;
        mCopyright = copyright != null ? copyright : "";
    }

    @NotNull
    public String getDate() {
        return mDate;
    }

    @NotNull
    public String getExplanation() {
        return mExplanation;
    }

    @NotNull
    public String getTitle() {
        return mTitle;
    }

    @NotNull
    public String getUrl() {
        return mUrl;
    }

    @NonNull
    public String getCopyright() {
        return mCopyright;
    }
}

File: data/src/main/java/com/example/data/model/APODRoom.kt
package com.example.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

/**
 * Pojo class для работы с Room
 */
@Entity
data class APODRoom(
        @field:ColumnInfo(name = "date") @field:PrimaryKey val date: String,
        @field:ColumnInfo(name = "explanation") val explanation: String,
        @field:ColumnInfo(name = "title") val title: String,
        @field:ColumnInfo(name = "url") val url: String,
        @field:ColumnInfo(name = "copyright") var copyright: String?) {
    init {
        this.copyright = copyright ?: ""
    }
}